kornia/augmentation/_2d/mix/transplantation.py

```diff
@@ -0,0 +1,261 @@
+          least some objects in the image are annotated.
+        - This implementation works for arbitrary spatial dimensions including 2D and 3D images.
+
+    Inputs:
+        - Input image tensors: :math:`(B, C, *)`.
+        - Segmentation mask tensors: :math:`(B, *)`.
+        - (optional) Additional image or mask tensors with are transformed in the same ways as the input image:
+          :math:`(B, C, *)` or :math:`(B, *)`.
+
+    Returns:
+        Union[Tuple[Tensor, Tensor], List[Tensor]]:
+
+        Tuple[Tensor, Tensor]:
+            - Augmented image tensors: :math:`(B, C, *)`.
+            - Augmented mask tensors: :math:`(B, *)`.
+        List[Tensor]:
+            - Augmented image tensors: :math:`(B, C, *)`.
+            - Augmented mask tensors: :math:`(B, *)`.
+            - Additional augmented image or mask tensors: :math:`(B, C, *)` or :math:`(B, *)`.
+
+    Examples:
+        >>> import torch
+        >>> rng = torch.manual_seed(0)
+        >>> aug = RandomTransplantation(p=1.)
+        >>> image = torch.randn(2, 3, 5, 5)
+        >>> mask = torch.randint(0, 3, (2, 5, 5))
+        >>> mask
+        tensor([[[0, 0, 1, 1, 0],
+                 [1, 2, 0, 0, 0],
+                 [1, 2, 1, 1, 0],
+                 [0, 0, 0, 0, 2],
+                 [2, 2, 2, 0, 2]],
+        <BLANKLINE>
+                [[2, 0, 0, 2, 1],
+                 [2, 1, 0, 2, 1],
+                 [2, 0, 1, 0, 2],
+                 [2, 2, 2, 0, 2],
+                 [2, 1, 0, 0, 0]]])
+        >>> image_out, mask_out = aug(image, mask)
+        >>> image_out.shape
+        torch.Size([2, 3, 5, 5])
+        >>> mask_out.shape
+        torch.Size([2, 5, 5])
+        >>> mask_out
+        tensor([[[2, 0, 1, 2, 0],
+                 [2, 2, 0, 2, 0],
+                 [2, 2, 1, 1, 2],
+                 [2, 2, 2, 0, 2],
+                 [2, 2, 2, 0, 2]],
+        <BLANKLINE>
+                [[0, 0, 0, 2, 0],
+                 [2, 1, 0, 0, 0],
+                 [2, 0, 1, 0, 0],
+                 [0, 0, 0, 0, 2],
+                 [2, 1, 0, 0, 0]]])
+
+    You can apply the same augmentation again, but the new mask should contain the same labels for the augmentation
+    to have an effect:
+
+        >>> aug._params["selected_labels"]  # Image 0 received label 2 from image 1 and image 1 label 0 from image 0
+        [tensor(2), tensor(0)]
+        >>> mask2 = torch.randint(0, 3, (2, 5, 5))
+        >>> image_out2, mask_out2 = aug(image, mask2, params=aug._params)
+        >>> mask_out2
+        tensor([[[2, 2, 2, 0, 1],
+                 [1, 2, 0, 1, 1],
+                 [0, 0, 1, 2, 2],
+                 [0, 0, 1, 2, 1],
+                 [1, 1, 1, 2, 0]],
+        <BLANKLINE>
+                [[0, 2, 2, 0, 0],
+                 [0, 0, 0, 0, 1],
+                 [0, 0, 0, 2, 0],
+                 [0, 0, 1, 0, 1],
+                 [1, 1, 1, 2, 0]]])
+    """
+
+    def __init__(
+        self,
+        excluded_labels: Optional[Sequence[int]] = None,
```

Comment:
```suggestion
        excluded_labels: Sequence[int] | None = None,
```
can you use typing annotations from `__future__` ?